### PR TITLE
ENT-1440. Fixed serializer exception when calculating unenroll and course start…

### PR DESCRIPTION
**Pull request by georgebabey**
… is None

### CODE REVIEW

**Comment by georgebabey** [comment]
Also ran make upgrade for some minor version bumps that I just kept in

**Review by brittneyexline — APPROVED** [review]

File: enterprise_data/api/v0/serializers.py [hunk 1]
```diff
@@ -29,7 +29,8 @@ def get_unenrollment_end_within_date(self, obj):
         """
         unenrollment_within_date = None
         if obj.unenrollment_timestamp:
-            unenrollment_within_date = (0 < (obj.unenrollment_timestamp - obj.course_start).days <= 14) or \
+            unenrollment_within_date = (not obj.course_start) or \
```
Comment (christopappas) [inline enterprise_data/api/v0/serializers.py]: Some style nitpicks, but, parentheses are not required to be around the `not obj.course_start`, although I can see parentheses making the longer expressions clearer (though I'm fairly certain they too are not necessary).

Also, I personally don't love backslashes. I usually evangelize use of parens as in, something like (untested):

```
unenrollment_within_date = (not obj.course_start or 
                                                (0 < (obj.unenrollment_timestamp - obj.course_start).days <= 14) or
                                                (0 < (obj.unenrollment_timestamp - obj.enrollment_created_timestamp).days <= 14))
```
or if those lines are too long:
```
unenrollment_within_date = (
    not obj.course_start or 
    0 < (obj.unenrollment_timestamp - obj.course_start).days <= 14 or
    0 < (obj.unenrollment_timestamp - obj.enrollment_created_timestamp).days <= 14
)
```

Just some suggestions anyway

**Review by christopappas — APPROVED** [review]
Looks good, just made a style suggestion in case you like that better

File: enterprise_data/api/v0/serializers.py (hunk 1 above)
Comment (georgebabey) [reply enterprise_data/api/v0/serializers.py]: Was just following the existing style, but agree and will update

**Pull request merged**